engine/src/simulation.cpp

```diff
@@ -1142,14 +1142,64 @@ StateId SimulationMachine::Reset::impl(SimulationContext& ctx) {
     }
     return true;
   });
+  ctx.sync.reset();
+  ctx.vehicles.clear();
+
+  // Create and configure all controllers:
+  for (const auto& c : ctx.config.vehicles) {
+    std::shared_ptr<cloe::Vehicle> v;
+    if (c.is_from_simulator()) {
+      auto& s = ctx.simulators.at(c.from_sim.simulator);
+      if (c.from_sim.is_by_name()) {
+        v = s->get_vehicle(c.from_sim.index_str);
+        if (!v) {
+          throw cloe::ModelError("simulator {} has no vehicle by name {}",
+                                 c.from_sim.simulator, c.from_sim.index_str)
+              .explanation("Simulator {} has following vehicles:\n{}",
+                           c.from_sim.simulator,
+                           enumerate_simulator_vehicles(*s));
+        }
+      } else {
+        v = s->get_vehicle(c.from_sim.index_num);
+        if (!v) {
+          throw cloe::ModelError("simulator {} has no vehicle at index {}",
+                                 c.from_sim.simulator, c.from_sim.index_num)
+              .explanation("Simulator {} has following vehicles:\n{}",
+                           c.from_sim.simulator,
+                           enumerate_simulator_vehicles(*s));
+        }
+      }
+      for (const auto& ctrl : ctx.config.controllers) {
+        if (ctrl.vehicle == c.name) {
+          auto name = ctrl.name.value_or(ctrl.binding);
+          assert(ctx.controllers.count(name) != 0);
+          ctx.controllers[name]->set_vehicle(v);
+        }
+      }
+      for (const auto& veh : ctx.config.vehicles) {
+        (void)veh;
+        // shall we reset ctx.vehicle from carmaker binding?
+      }
+    } else {
+      logger()->info("Shall we reset veh from vehicles ?");
+      //   if (ctx.vehicles.count(c.from_veh)) {
+      //     v = ctx.vehicles.at(c.from_veh);
+      //   } else {
+      //     // This vehicle depends on another that hasn't been create yet.
+      //     return nullptr;
+      //   }
+      // }
+    }
+  }
+
   if (ok) {
-    return CONNECT;
+    return START;
```

Comment:
- [ ] Does this match the state transition diagram?

If yes, then we are done, otherwise we need to update the diagrm.

Alternatively, we should ask:

- [ ] Why not `CONNECT` state?
 
If a participant is already connected, it could just be a no-op. Other thought: we already have `RESET` so that takes the place of `CONNECT` maybe?

- [ ] Is there anything happening in `CONNECT` that needs to happen in a `RESET`?

Reply:
**_Does this match the state transition diagram?_**
I think so, because from the diagram in comment , reset should jump to START instead of CONNECT ,
https://github.com/eclipse/cloe/blob/10737bfd719c98edbf53c70e92513d32c6417ddf/engine/src/simulation.cpp#L60
And from semantic understanding, we do not want and have to retry "CONNECT" for each participant after reset happened, because some operation may be rather time-costing like initializing the target binding(simulator/controller), and some operation may be even not allowed, just like we could only connect the same socket port once.

**_Is there anything happening in CONNECT that needs to happen in a RESET?_**
I think no, but in connect impl many operations is not allowed to try multiple times, like register triggers and events, right? 

Generatlly, for current code implementation, I think START is better choice to get proposal RESTART result.